### PR TITLE
refactor: Migrate Node dropdown menu to Compose

### PR DESCRIPTION
**Pull request by jamesarich**
Migrated the Nodes dropdown menu to Compose, replacing the old XML-based UI. This change improves the maintainability and flexibility of the UI, allowing for more dynamic and interactive elements. It also simplifies the codebase by reducing the need for separate XML menu.

The refactoring includes:

- Removed XML menu and related code.
- Created composable functions for NodeMenu and MenuItemAction.
- Updated SimpleAlertDialog to support confirmation action.
- Updated NodeItem to use the new NodeMenu and MenuItemAction.
- Updated UsersFragment to handle NodeMenu actions and navigation.

[Screen_recording_20241106_192131.webm](https://github.com/user-attachments/assets/fabce572-9790-4d03-b16c-2b30eafa8fa7)

### CODE REVIEW

File: app/src/main/java/com/geeksville/mesh/ui/components/NodeMenu.kt [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+            )
+            Divider()
+            DropdownMenuItem(
+                onClick = {
+                    onDismissRequest()
+                    onMenuItemAction(MenuItemAction.Ignore)
+                },
+                content = {
+                    Row {
+                        Checkbox(
```
Comment (andrekir) [inline app/src/main/java/com/geeksville/mesh/ui/components/NodeMenu.kt]: - `DropdownMenuItem` content is already RowScope, so Row isn't needed.
- Checkbox should come after the text.

File: app/src/main/java/com/geeksville/mesh/ui/components/NodeMenu.kt [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+                content = { Text(stringResource(R.string.traceroute)) }
+            )
+            Divider()
+            DropdownMenuItem(
+                onClick = {
+                    onDismissRequest()
+                    onMenuItemAction(MenuItemAction.Remove)
+                },
+                content = { Text(stringResource(R.string.remove)) },
+                enabled = ignoreIncomingList.size < 3 || isIgnored
```
Comment (andrekir) [inline app/src/main/java/com/geeksville/mesh/ui/components/NodeMenu.kt]: copy-paste typo?

File: app/src/main/java/com/geeksville/mesh/ui/components/NodeMenu.kt [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+    ignoreIncomingList: List<Int>,
+    isThisNode: Boolean = false,
+    onMenuItemAction: (MenuItemAction) -> Unit,
+    onDismissRequest: () -> Unit,
+    expanded: Boolean = false,
+) {
+    val isIgnored = ignoreIncomingList.contains(node.num)
+
+    DropdownMenu(
+        modifier = Modifier.Companion.background(MaterialTheme.colors.background.copy(alpha = 1f)),
```
Comment (andrekir) [inline app/src/main/java/com/geeksville/mesh/ui/components/NodeMenu.kt]: Companion?

File: app/src/main/java/com/geeksville/mesh/ui/UsersFragment.kt [hunk 4]
```diff
@@ -142,6 +103,49 @@ fun NodesScreen(
...
+                displayIgnoreDialogForNode?.let {
+                    model.removeNode(it.num)
+                }
+                displayRemoveDialogForNode = null
+            },
+            onDismiss = {
+                displayRemoveDialogForNode = null
+            }
+        )
+    }
```
Comment (andrekir) [inline app/src/main/java/com/geeksville/mesh/ui/UsersFragment.kt]: can this block move inside the NodeMenu component?

File: app/src/main/java/com/geeksville/mesh/ui/UsersFragment.kt [hunk 5]
```diff
@@ -37,56 +43,6 @@ class UsersFragment : ScreenFragment("Users"), Logging {
 
     private val model: UIViewModel by activityViewModels()
 
-    private fun popup(node: NodeEntity) {
-        if (!model.isConnected()) return
```
Comment (andrekir) [inline app/src/main/java/com/geeksville/mesh/ui/UsersFragment.kt:41]: we're missing a similar check on the refactored code. except for "more details" the other actions only work while connected.

**Review by andrekir — COMMENTED** [review]
thanks for this. overall looks good, added a few comments before we merge.

**Comment by jamesarich** [comment]
Thanks for the review, was noticing that the dropdown wasn't anchored to the chip and went on a refactoring spree 😅 . I'll get it cleaned up.

File: app/src/main/java/com/geeksville/mesh/ui/UsersFragment.kt (hunk 4 above)
Comment (jamesarich) [reply app/src/main/java/com/geeksville/mesh/ui/UsersFragment.kt]: I was receiving crashes when this was contained in the menu.. I'll give it another shot.

File: app/src/main/java/com/geeksville/mesh/ui/components/NodeMenu.kt (hunk 2 above)
Comment (jamesarich) [reply app/src/main/java/com/geeksville/mesh/ui/components/NodeMenu.kt]: i was attempting to replicate the `enabled` logic contained [here](https://github.com/meshtastic/Meshtastic-Android/pull/1386/files#diff-198b8c397de541c4449058d13639ab9cd0b56c3cedd98c99619983819d8b304aL22)

File: app/src/main/java/com/geeksville/mesh/ui/components/NodeMenu.kt (hunk 3 above)
Comment (jamesarich) [reply app/src/main/java/com/geeksville/mesh/ui/components/NodeMenu.kt]: Gemini likes to add that in autocomplete, for some reason.. missed this one

**Comment by jamesarich** [comment]
As per discord discussion:
>using "isConnected = model.isConnected()" wont trigger recompositions when connection state changes
it needs to collect the state directly or in the nodesUiState

**Pull request merged**